client/src/NationalStats.tsx

```diff
@@ -0,0 +1,55 @@
+import { Component } from 'react';
+import CanadaNationalStats from './Models/CanadaNationalStats';
+
+export default class NationalStats extends Component<{longitude: any, latitude: any}, {canadaNationalStats: CanadaNationalStats}> {
+    async componentDidMount(): Promise<void> {
+    }
+    async componentWillReceiveProps(nextProps: {latitude: any, longitude: any}): Promise<void> {
+        if (nextProps.latitude !== 0 && nextProps.longitude !== 0) {
+            this.setState({canadaNationalStats: await this.getLocationFromPosition(nextProps.latitude, nextProps.longitude)});
+        }
+    }
+    componentWillUnmount() {
+        // fix Warning: Can't perform a React state update on an unmounted component
+        this.setState = (state,callback)=>{
+            return;
+        };
+    }
+
+    async getLocationFromPosition(latitude: number, longitude: number): Promise<any> {
+
+        try {
+            const url = `http://localhost:8080/location?longitude=${longitude.toPrecision(5)}&latitude=${latitude.toPrecision(5)}&component=nationalstats`;
```

Comment:
Look into rest notation, this is a poorly worded url. Should probably be something like "getNationalStatsByCoords?latitude=*longitude=*". Essentially, we should be using a request parameter to act as a switch for different funcitonality, that should be different endpoints.